### ISSUE TITLE
[Phase 7] Full Pipeline and Documentation

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/neuronal-spike-sorting#8]
## Objective

Create end-to-end spike sorting pipeline with CLI interface.

## Dependencies

- All previous phases completed

## Tasks

- [ ] Implement `src/pipeline.py` with full sorting pipeline
- [ ] Implement `scripts/sort_recording.py` CLI entry point
- [ ] Create `requirements.txt` with all dependencies
- [ ] Benchmark on SpikeForest datasets
- [ ] Ensure reproducibility
- [ ] Update README with usage instructions
- [ ] Final testing and validation

## Files to Create

| File | Purpose |
|------|---------|
| `src/pipeline.py` | End-to-end pipeline |
| `scripts/sort_recording.py` | CLI interface |
| `requirements.txt` | Dependencies |

## Starter Code

```python
# src/pipeline.py
"""End-to-end spike sorting pipeline."""

import numpy as np
from pathlib import Path

from src.data.recording_loader import load_recording
from src.preprocessing.filters import bandpass_filter, common_average_reference
from src.detection.threshold_detector import detect_spikes
from src.detection.waveform_extractor import extract_waveforms
from src.features.pca_features import WaveformPCA
from src.clustering.hdbscan_clustering import HDBSCANSorter

class SpikeSortingPipeline:
    """Full spike sorting pipeline."""

    def __init__(self, config: dict = None):
        self.config = config or self._default_config()

    def _default_config(self) -> dict:
        return {
            "filter_low": 300,
            "filter_high": 6000,
            "threshold": -5.0,
            "n_pca_components": 10,
            "min_cluster_size": 20,
        }

    def run(self, recording_path: str) -> dict:
        """Run full sorting pipeline."""
        # Load recording
        rec = load_recording(recording_path)
        data = rec.get_traces().T  # (n_channels, n_samples)
        fs = rec.get_sampling_frequency()

        # Preprocess
        filtered = bandpass_filter(
            data, fs,
            low=self.config["filter_low"],
            high=self.config["filter_high"]
        )
        filtered = common_average_reference(filtered)

        # Detect spikes
        spike_times = detect_spikes(
            filtered, fs,
            threshold=self.config["threshold"]
        )

        # Extract waveforms
        waveforms = extract_waveforms(filtered, spike_times, fs)

        # Feature extraction
        pca = WaveformPCA(n_components=self.config["n_pca_components"])
        features = pca.fit_transform(waveforms)

        # Clustering
        sorter = HDBSCANSorter(
            min_cluster_size=self.config["min_cluster_size"]
        )
        labels = sorter.fit_predict(features)

        return {
            "spike_times": spike_times,
            "labels": labels,
            "waveforms": waveforms,
            "features": features,
        }
```

```python
# scripts/sort_recording.py
"""CLI for spike sorting."""

import argparse
import numpy as np
from src.pipeline import SpikeSortingPipeline

def main():
    parser = argparse.ArgumentParser(description="Spike Sorting Pipeline")
    parser.add_argument("recording", help="Path to recording")
    parser.add_argument("-o", "--output", default="sorting_results.npz")
    parser.add_argument("--threshold", type=float, default=-5.0)

    args = parser.parse_args()

    config = {"threshold": args.threshold}
    pipeline = SpikeSortingPipeline(config)
    results = pipeline.run(args.recording)

    np.savez(args.output, **results)
    print(f"Sorted {len(np.unique(results['labels'])) - 1} units")
    print(f"Results saved to {args.output}")

if __name__ == "__main__":
    main()
```

## Definition of Done

- [ ] Single command sorts recording
- [ ] Results reproducible with seed
- [ ] Benchmarked on SpikeForest datasets
- [ ] Documentation complete
- [ ] All tests passing
- [ ] README updated with usage examples

---
Part of #1 (Meta Issue)
